src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -153,6 +153,74 @@ TEST_CASE("CheckCopyMovingVanillaNetworkTest", "[FeedForwardNetworkTest]")
   CheckMoveFunction<>(model1, trainData, trainLabels, 1);
 }
 
+/**
+ * Check whether copying and moving network with AdaptiveMaxPooling is working or not.
+ */
+
+TEST_CASE("CheckCopyMovingAdaptiveMaxPoolingNetworkTest",
```

Comment:
Did you want to add a test for `AddMerge` too?

Reply:
Yea, I will add a test for `AddMerge`, my bad I forgot about that